cogs/utility.py

```diff
@@ -1432,6 +1440,188 @@ async def permissions_override(self, ctx, command_name: str.lower, *, level_name
+                    "message",
+                    check=lambda m: m.author == ctx.author and m.channel == ctx.channel,
+                    timeout=60.0,
+                )
+                if msg.content.lower() not in ("y", "yes"):
+                    return await ctx.send(
+                        embed=discord.Embed(
+                            title="Operation Aborted",
+                            description="No changes have been applied.",
+                            color=self.bot.error_color,
+                        )
+                    )
+            except asyncio.TimeoutError:
+                return await ctx.send(
+                    embed=discord.Embed(title="Error", description="Timed out.", color=self.bot.error_color)
+                )
+
+        if not found_commands:
+            return await ctx.send(
+                embed=discord.Embed(
+                    title="Error",
+                    description="No valid commands provided. Aborting.",
+                    color=self.bot.error_color,
+                )
+            )
+
+        # Expand subcommands
+        final_commands = set()
+
+        def add_command_recursive(cmd):
+            final_commands.add(cmd)
+            if hasattr(cmd, "commands"):
+                for sub in cmd.commands:
+                    add_command_recursive(sub)
+
+        for cmd in found_commands:
+            add_command_recursive(cmd)
+
+        embed = discord.Embed(
+            title="Select Permission Level",
+            description=(
+                f"Found {len(final_commands)} commands (including subcommands).\n"
+                "What permission level should these commands be set to? (e.g. `Owner`, `Admin`, `Moderator`, `Supporter`, `User`)"
+            ),
+            color=self.bot.main_color,
+        )
+        await ctx.send(embed=embed)
+
+        try:
+            msg = await self.bot.wait_for(
+                "message",
+                check=lambda m: m.author == ctx.author and m.channel == ctx.channel,
+                timeout=60.0,
+            )
+
+        except asyncio.TimeoutError:
+            return await ctx.send(
+                embed=discord.Embed(title="Error", description="Timed out.", color=self.bot.error_color)
+            )
+
+        level_name = msg.content
+        level = self._parse_level(level_name)
+        if level == PermissionLevel.INVALID:
+            return await ctx.send(
+                embed=discord.Embed(
+                    title="Error",
+                    description=f"Invalid permission level: `{level_name}`. Aborting.",
+                    color=self.bot.error_color,
+                )
+            )
+
+        # Confirmation
+        command_list_str = ", ".join(
+            f"`{c.qualified_name}`" for c in sorted(final_commands, key=lambda x: x.qualified_name)
+        )
+
+        # Truncate if too long for embed description
+        if len(command_list_str) > 2000:
```

Comment:
There is a `return_or_truncate(text, max_len)` function in core/utils.py that does this.